docs/DeveloperGuide.md

```diff
@@ -149,6 +149,39 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Theme
+
+The implementation of theme is done by parsing the raw values of the `json` file containing a color palette, and then
+transforming them into a `css` file that is then applied to `MainWindow.fxml`. All other elements will read from the
+same `css` file. When the application is first launched, it will apply the default theme constructed by `ThemeFactory`.
+When a theme is set by the command, it and it's temporary `CSS` file will be stored as variables in `ThemeManager`,
+which will subsequently be used by the other parts of the application (for e.g. messageboxes).
+
+####Initialization
+
+When the application starts up, it first checks for any previously used themes in `UserPrefs`. If not found, the
+application continues to use the default theme. Otherwise, it attempts to load the theme file with
+`ThemeFactory#load()`.
+
+![Sequence diagram for `ThemeManager` initialization](images/ThemeInitializationSequenceDiagram.png)
+
+####Command Invocation
```

Comment:
Do add a space between the hashes and the title itself, as I'm not sure if jekyll can parse the markdown properly without